fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/api/event/resource/PackProvisionCallback.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.resource;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.client.resource.ClientResourcePackContainer;
+import net.minecraft.resource.ResourcePackContainer;
+import net.minecraft.resource.ResourcePackContainerManager;
+
+/**
+ * Callback for registering custom resource pack providers.
+ *
+ * @param <T> the resource pack container type
+ */
+public interface PackProvisionCallback<T extends ResourcePackContainer> {
```

Comment:
`@FunctionalInterface`

Reply:
That's optional and does not affect lambda usage.

Reply:
Fixed